app/services/arcgis_api/geocoder.rb

```diff
@@ -94,6 +94,8 @@ def faraday
 
         # Parse JSON responses
         conn.response :json, content_type: 'application/json'
+
+        custom_connection_options&.call(conn)
```

Comment:
this is arguably "more ruby" but "less clear to read"... even use the magic `yield if block_given?` syntax and skip naming the block at all 😱 

```suggestion
        yield conn if block_given?
```

(we could remove the `&custom_connection_options argument entirely so I'd at least liek add a  `# @yield ` yard comment as a hint